integration/tests/cook/test_basic.py

```diff
@@ -232,10 +235,10 @@ def test_multiple_progress_updates_submit(self):
 
     def test_multiple_rapid_progress_updates_submit(self):
         job_executor_type = util.get_job_executor_type(self.cook_url)
-        command = ''.join(['echo "progress: {0} {0}%" && '.format(a) for a in range(1, 100, 4)]) + \
-                  ''.join(['echo "progress: {0} {0}%" && '.format(a) for a in range(99, 40, -4)]) + \
-                  ''.join(['echo "progress: {0} {0}%" && '.format(a) for a in range(40, 81, 2)]) + \
-                  'echo "Done" && exit 0'
+        def progress_string(a):
+            return util.output_progress_string(self.cook_url, a, f'{a}%')
```

Comment:
I would favor a blank line before and after the `def`

Reply:
Done.